From 9ac47a43ffc8e3c73867b9953145a8e7a227132d Mon Sep 17 00:00:00 2001
From: Toni Spets <toni.spets@iki.fi>
Date: Wed, 6 May 2015 14:18:34 +0100
Subject: [PATCH] Skip discarded resource sections when building a PE resource
 table.

	PR ld/18372
	* peXXigen.c (rsrc_process_section): Skip discarded resource
	sections.
---
 bfd/ChangeLog  | 6 ++++++
 bfd/peXXigen.c | 3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/bfd/peXXigen.c b/bfd/peXXigen.c
index 6aa18ca..c897c64 100644
--- a/bfd/peXXigen.c
+++ b/bfd/peXXigen.c
@@ -4160,7 +4160,8 @@ rsrc_process_section (bfd * abfd,
     {
       asection * rsrc_sec = bfd_get_section_by_name (input, ".rsrc");
 
-      if (rsrc_sec != NULL)
+      /* PR 18372 - skip discarded .rsrc sections.  */
+      if (rsrc_sec != NULL && !discarded_section (rsrc_sec))
 	{
 	  if (num_input_rsrc == max_num_input_rsrc)
 	    {
-- 
2.1.0

